standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -4013,6 +4016,12 @@ private String getShowCompactFilterClause(ShowCompactRequest request) {
       " WHERE " + StringUtils.join(" AND ", params) : EMPTY;
   }
 
+  private String getShowCompactSortingOrderClause(ShowCompactRequest request){
+   String sortingOrder = request.getOrder();
+   return  isNotBlank(sortingOrder) ? TxnQueries.SHOW_COMPACTION_ORDERBY_CLAUSE + " , " + sortingOrder :
```

Comment:
if custom order is provided we should use that and don't append the default one

Reply:
fixed